src/main/java/utilities/TaskList.java

```diff
@@ -0,0 +1,122 @@
+package utilities;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import exceptions.WilliamException;
+import tasks.Task;
+
+/**
+ * Contains the task list e.g. it has operations to add/delete tasks in the list
+ */
+public class TaskList {
+    private List<Task> tasks;
+
+    /**
+     * Constructor if there is no existing file
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Constructor if there is an existing txt file with tasks
+     */
+    public TaskList(List<Task> tasks) {
```

Comment:
Would it be more specific to constrain the List type to ArrayList<Task>? Instead of the abstract class. This would prevent other List data structures like Queue/Stack to be passed into the function, which could interfere with the intended outcome of any add/remove/search methods you have.